### PR TITLE
🌱 Add a function to retrieve nodes using labels

### PR DESCRIPTION
**Pull request by Xenwar**
Add a function to retrieve nodes using labels.
Needed for avoiding code duplication as nodes will be retrieved using new and legacy labels of the providerID.

Required by: https://github.com/metal3-io/cluster-api-provider-metal3/pull/324

### CODE REVIEW

**Review by kashifest — APPROVED** [review]

**Comment by fmuyassarov** [comment]
/retitle 🌱 Add a function to retrieve nodes using labels

File: baremetal/metal3machine_manager.go [hunk 1]
```diff
@@ -1620,3 +1619,23 @@ func (m *MachineManager) getMachineSet(ctx context.Context) (*capi.MachineSet, e
 	}
 	return nil, errors.New("MachineSet is not found")
 }
+
+// getNodeWithLabel gets kubernetes nodes with a given label
```
Comment (furkatgofurov7) [inline baremetal/metal3machine_manager.go:1623]: /s
```suggestion
// getNodesWithLabel gets kubernetes nodes with a given label
```

**Review by furkatgofurov7 — COMMENTED** [review]
/hold
/lgtm

This looks good, small nit but not a blocker, can be fixed in the same patch or in follow up or even in #324. Feel free to unhold if you think this should go in as is :)

**Review by kashifest — APPROVED** [review]
Please address @furkatgofurov7's comment

File: baremetal/metal3machine_manager.go (hunk 1 above)
Comment (Xenwar) [reply baremetal/metal3machine_manager.go]: Thanks. fixed.

**Pull request merged**